src/framework/utils.hpp

```diff
@@ -20,6 +20,9 @@
 #include <cmath>
 #include <limits>
 
+#include <immintrin.h>
+#include <type_traits>
```

Comment:
These don't seem to be used in this header file so I don't think they should be here.

If they are put here for including in another file these seems dangerous since it is not obvious where they will be loaded from.